ResultIterator (#176)
thecodingmachine/tdbm: doc/quickstart.md

![Database schema](images/schema1.png)

DAOs, Beans and ResultIterators
--------------

When you [installed TDBM](install.md), you spent some time configuring a connection to your database and then, you generated the DAOs, beans and resultIterators.

> There will be roughly **one class (DAO, Bean, ResultIterator) per table**, in fact only ManyToMany link tables won't get their class.

Reviewer: ```suggestion
There will be roughly **one class (DAO, Bean, ResultIterator) per table**, in fact only ManyToMany link tables won't get their classes.
```

Each time your database model changes (if you add a new table, a new column, an index or a foreign key...), you will need to regenerate those DAOs, beans and resultIterators.

### DAOs

DAOs are "Data Access Objects". DAOs are classes that will help you access the objects in your database.
Each DAO will return a Bean or a ResultIterator (collection of Beans).

### Beans

Each row in your database will be represented by one instance of a bean.
This instance will allow you to easily update your data using setters and getters strongly typed by TDBM or created by you.

### ResultIterators

> A ResultIterator will represent the result of a query. It will allow you to specify it.

Reviewer: ```suggestion
A ResultIterator object represents the result of a query. It is an iterable object. You can use it to paginate through the results, sort the results or (in the future) filter the results.
```

  column of the "users" table. It is generated by TDBM and you should never modify this class.
- `User`: this class extends `AbstractUser`. If you have some custom getters and setters, you should implement them in this class. You can
  edit it as TDBM will never overwrite it.

> - `AbstractUserResultIterator`: the resultIterator representing a query on the "users" table. This class contains methods to specify you query.

Reviewer: ```suggestion
- `AbstractUserResultIterator`: the resultIterator representing a query on the "users" table. This class contains methods to filter/sort/paginate your query.
```
